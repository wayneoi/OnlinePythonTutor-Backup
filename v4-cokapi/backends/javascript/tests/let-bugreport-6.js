// https://github.com/pgbovine/OnlinePythonTutor/issues/192
// see below for user report ...
let categoryList = ['tiger', 'lion', 'house']
  
categoryList.forEach(c => {
  console.log('outside of if', c)
})

if (categoryList){
  // user: "changing the let to a var on [the next line] makes the script work as expected."
  let categories = 'our categories are:'
  
  console.log('inside of the if')
  
  categoryList.forEach(c => {
    console.log('in if', c)
    categories += `${c}, `
  })
  
  console.log(categories)
}
