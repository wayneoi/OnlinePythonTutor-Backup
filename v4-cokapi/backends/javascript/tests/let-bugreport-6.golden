{ code: '// https://github.com/pgbovine/OnlinePythonTutor/issues/192\n// see below for user report ...\nlet categoryList = [\'tiger\', \'lion\', \'house\']\n  \ncategoryList.forEach(c => {\n  console.log(\'outside of if\', c)\n})\n\nif (categoryList){\n  // user: "changing the let to a var on [the next line] makes the script work as expected."\n  let categories = \'our categories are:\'\n  \n  console.log(\'inside of the if\')\n  \n  categoryList.forEach(c => {\n    console.log(\'in if\', c)\n    categories += `${c}, `\n  })\n  \n  console.log(categories)\n}',
  trace: 
   [ { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: {},
       ordered_globals: [],
       line: 3,
       col: 0,
       event: 'step_line',
       heap: {} },
     { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: { categoryList: [ 'REF', 1 ] },
       ordered_globals: [ 'categoryList' ],
       line: 5,
       col: 0,
       event: 'step_line',
       heap: { '1': [ 'LIST', 'tiger', 'lion', 'house' ] } },
     { stdout: '',
       func_name: '',
       stack_to_render: 
        [ { func_name: '',
            frame_id: 2,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: '_f2',
            ordered_varnames: [ 'c' ],
            encoded_locals: { c: 'tiger' } } ],
       globals: { categoryList: [ 'REF', 1 ] },
       ordered_globals: [ 'categoryList' ],
       line: 6,
       col: 2,
       event: 'call',
       heap: { '1': [ 'LIST', 'tiger', 'lion', 'house' ] } },
     { stdout: 'outside of if tiger\n',
       func_name: '',
       stack_to_render: 
        [ { func_name: '',
            frame_id: 2,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: '_f2',
            ordered_varnames: [ 'c', '__return__' ],
            encoded_locals: { c: 'tiger', __return__: [ 'JS_SPECIAL_VAL', 'undefined' ] } } ],
       globals: { categoryList: [ 'REF', 1 ] },
       ordered_globals: [ 'categoryList' ],
       line: 6,
       col: 0,
       event: 'return',
       heap: { '1': [ 'LIST', 'tiger', 'lion', 'house' ] } },
     { stdout: 'outside of if tiger\n',
       func_name: '',
       stack_to_render: 
        [ { func_name: '',
            frame_id: 3,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: '_f3',
            ordered_varnames: [ 'c' ],
            encoded_locals: { c: 'lion' } } ],
       globals: { categoryList: [ 'REF', 1 ] },
       ordered_globals: [ 'categoryList' ],
       line: 6,
       col: 2,
       event: 'step_line',
       heap: { '1': [ 'LIST', 'tiger', 'lion', 'house' ] } },
     { stdout: 'outside of if tiger\noutside of if lion\n',
       func_name: '',
       stack_to_render: 
        [ { func_name: '',
            frame_id: 3,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: '_f3',
            ordered_varnames: [ 'c', '__return__' ],
            encoded_locals: { c: 'lion', __return__: [ 'JS_SPECIAL_VAL', 'undefined' ] } } ],
       globals: { categoryList: [ 'REF', 1 ] },
       ordered_globals: [ 'categoryList' ],
       line: 6,
       col: 0,
       event: 'return',
       heap: { '1': [ 'LIST', 'tiger', 'lion', 'house' ] } },
     { stdout: 'outside of if tiger\noutside of if lion\n',
       func_name: '',
       stack_to_render: 
        [ { func_name: '',
            frame_id: 4,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: '_f4',
            ordered_varnames: [ 'c' ],
            encoded_locals: { c: 'house' } } ],
       globals: { categoryList: [ 'REF', 1 ] },
       ordered_globals: [ 'categoryList' ],
       line: 6,
       col: 2,
       event: 'step_line',
       heap: { '1': [ 'LIST', 'tiger', 'lion', 'house' ] } },
     { stdout: 'outside of if tiger\noutside of if lion\noutside of if house\n',
       func_name: '',
       stack_to_render: 
        [ { func_name: '',
            frame_id: 4,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: '_f4',
            ordered_varnames: [ 'c', '__return__' ],
            encoded_locals: { c: 'house', __return__: [ 'JS_SPECIAL_VAL', 'undefined' ] } } ],
       globals: { categoryList: [ 'REF', 1 ] },
       ordered_globals: [ 'categoryList' ],
       line: 6,
       col: 0,
       event: 'return',
       heap: { '1': [ 'LIST', 'tiger', 'lion', 'house' ] } },
     { stdout: 'outside of if tiger\noutside of if lion\noutside of if house\n',
       func_name: '',
       stack_to_render: [],
       globals: { categoryList: [ 'REF', 1 ] },
       ordered_globals: [ 'categoryList' ],
       line: 9,
       col: 0,
       event: 'step_line',
       heap: { '1': [ 'LIST', 'tiger', 'lion', 'house' ] } },
     { stdout: 'outside of if tiger\noutside of if lion\noutside of if house\n',
       func_name: '',
       stack_to_render: [],
       globals: { categoryList: [ 'REF', 1 ] },
       ordered_globals: [ 'categoryList' ],
       line: 11,
       col: 2,
       event: 'step_line',
       heap: { '1': [ 'LIST', 'tiger', 'lion', 'house' ] } },
     { stdout: 'outside of if tiger\noutside of if lion\noutside of if house\n',
       func_name: '',
       stack_to_render: [],
       globals: 
        { categoryList: [ 'REF', 1 ],
          'categories (block 1)': 'our categories are:' },
       ordered_globals: [ 'categoryList', 'categories (block 1)' ],
       line: 13,
       col: 2,
       event: 'step_line',
       heap: { '1': [ 'LIST', 'tiger', 'lion', 'house' ] } },
     { stdout: 'outside of if tiger\noutside of if lion\noutside of if house\ninside of the if\n',
       func_name: '',
       stack_to_render: [],
       globals: 
        { categoryList: [ 'REF', 1 ],
          'categories (block 1)': 'our categories are:' },
       ordered_globals: [ 'categoryList', 'categories (block 1)' ],
       line: 15,
       col: 2,
       event: 'step_line',
       heap: { '1': [ 'LIST', 'tiger', 'lion', 'house' ] } },
     { stdout: 'outside of if tiger\noutside of if lion\noutside of if house\ninside of the if\n',
       func_name: '',
       stack_to_render: 
        [ { func_name: '',
            frame_id: 5,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: '_f5',
            ordered_varnames: [ 'categories (block 0)', 'c' ],
            encoded_locals: { 'categories (block 0)': 'our categories are:', c: 'tiger' } } ],
       globals: { categoryList: [ 'REF', 1 ] },
       ordered_globals: [ 'categoryList' ],
       line: 16,
       col: 4,
       event: 'call',
       heap: { '1': [ 'LIST', 'tiger', 'lion', 'house' ] } },
     { stdout: 'outside of if tiger\noutside of if lion\noutside of if house\ninside of the if\nin if tiger\n',
       func_name: '',
       stack_to_render: 
        [ { func_name: '',
            frame_id: 5,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: '_f5',
            ordered_varnames: [ 'categories (block 0)', 'c' ],
            encoded_locals: { 'categories (block 0)': 'our categories are:', c: 'tiger' } } ],
       globals: { categoryList: [ 'REF', 1 ] },
       ordered_globals: [ 'categoryList' ],
       line: 17,
       col: 4,
       event: 'step_line',
       heap: { '1': [ 'LIST', 'tiger', 'lion', 'house' ] } },
     { stdout: 'outside of if tiger\noutside of if lion\noutside of if house\ninside of the if\nin if tiger\n',
       func_name: '',
       stack_to_render: 
        [ { func_name: '',
            frame_id: 5,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: '_f5',
            ordered_varnames: [ 'categories (block 0)', 'c', '__return__' ],
            encoded_locals: 
             { 'categories (block 0)': 'our categories are:tiger, ',
               c: 'tiger',
               __return__: [ 'JS_SPECIAL_VAL', 'undefined' ] } } ],
       globals: { categoryList: [ 'REF', 1 ] },
       ordered_globals: [ 'categoryList' ],
       line: 17,
       col: 2,
       event: 'return',
       heap: { '1': [ 'LIST', 'tiger', 'lion', 'house' ] } },
     { stdout: 'outside of if tiger\noutside of if lion\noutside of if house\ninside of the if\nin if tiger\n',
       func_name: '',
       stack_to_render: 
        [ { func_name: '',
            frame_id: 6,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: '_f6',
            ordered_varnames: [ 'categories (block 0)', 'c' ],
            encoded_locals: 
             { 'categories (block 0)': 'our categories are:tiger, ',
               c: 'lion' } } ],
       globals: { categoryList: [ 'REF', 1 ] },
       ordered_globals: [ 'categoryList' ],
       line: 16,
       col: 4,
       event: 'step_line',
       heap: { '1': [ 'LIST', 'tiger', 'lion', 'house' ] } },
     { stdout: 'outside of if tiger\noutside of if lion\noutside of if house\ninside of the if\nin if tiger\nin if lion\n',
       func_name: '',
       stack_to_render: 
        [ { func_name: '',
            frame_id: 6,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: '_f6',
            ordered_varnames: [ 'categories (block 0)', 'c' ],
            encoded_locals: 
             { 'categories (block 0)': 'our categories are:tiger, ',
               c: 'lion' } } ],
       globals: { categoryList: [ 'REF', 1 ] },
       ordered_globals: [ 'categoryList' ],
       line: 17,
       col: 4,
       event: 'step_line',
       heap: { '1': [ 'LIST', 'tiger', 'lion', 'house' ] } },
     { stdout: 'outside of if tiger\noutside of if lion\noutside of if house\ninside of the if\nin if tiger\nin if lion\n',
       func_name: '',
       stack_to_render: 
        [ { func_name: '',
            frame_id: 6,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: '_f6',
            ordered_varnames: [ 'categories (block 0)', 'c', '__return__' ],
            encoded_locals: 
             { 'categories (block 0)': 'our categories are:tiger, lion, ',
               c: 'lion',
               __return__: [ 'JS_SPECIAL_VAL', 'undefined' ] } } ],
       globals: { categoryList: [ 'REF', 1 ] },
       ordered_globals: [ 'categoryList' ],
       line: 17,
       col: 2,
       event: 'return',
       heap: { '1': [ 'LIST', 'tiger', 'lion', 'house' ] } },
     { stdout: 'outside of if tiger\noutside of if lion\noutside of if house\ninside of the if\nin if tiger\nin if lion\n',
       func_name: '',
       stack_to_render: 
        [ { func_name: '',
            frame_id: 7,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: '_f7',
            ordered_varnames: [ 'categories (block 0)', 'c' ],
            encoded_locals: 
             { 'categories (block 0)': 'our categories are:tiger, lion, ',
               c: 'house' } } ],
       globals: { categoryList: [ 'REF', 1 ] },
       ordered_globals: [ 'categoryList' ],
       line: 16,
       col: 4,
       event: 'step_line',
       heap: { '1': [ 'LIST', 'tiger', 'lion', 'house' ] } },
     { stdout: 'outside of if tiger\noutside of if lion\noutside of if house\ninside of the if\nin if tiger\nin if lion\nin if house\n',
       func_name: '',
       stack_to_render: 
        [ { func_name: '',
            frame_id: 7,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: '_f7',
            ordered_varnames: [ 'categories (block 0)', 'c' ],
            encoded_locals: 
             { 'categories (block 0)': 'our categories are:tiger, lion, ',
               c: 'house' } } ],
       globals: { categoryList: [ 'REF', 1 ] },
       ordered_globals: [ 'categoryList' ],
       line: 17,
       col: 4,
       event: 'step_line',
       heap: { '1': [ 'LIST', 'tiger', 'lion', 'house' ] } },
     { stdout: 'outside of if tiger\noutside of if lion\noutside of if house\ninside of the if\nin if tiger\nin if lion\nin if house\n',
       func_name: '',
       stack_to_render: 
        [ { func_name: '',
            frame_id: 7,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: '_f7',
            ordered_varnames: [ 'categories (block 0)', 'c', '__return__' ],
            encoded_locals: 
             { 'categories (block 0)': 'our categories are:tiger, lion, house, ',
               c: 'house',
               __return__: [ 'JS_SPECIAL_VAL', 'undefined' ] } } ],
       globals: { categoryList: [ 'REF', 1 ] },
       ordered_globals: [ 'categoryList' ],
       line: 17,
       col: 2,
       event: 'return',
       heap: { '1': [ 'LIST', 'tiger', 'lion', 'house' ] } },
     { stdout: 'outside of if tiger\noutside of if lion\noutside of if house\ninside of the if\nin if tiger\nin if lion\nin if house\n',
       func_name: '',
       stack_to_render: [],
       globals: 
        { categoryList: [ 'REF', 1 ],
          'categories (block 1)': 'our categories are:tiger, lion, house, ' },
       ordered_globals: [ 'categoryList', 'categories (block 1)' ],
       line: 20,
       col: 2,
       event: 'step_line',
       heap: { '1': [ 'LIST', 'tiger', 'lion', 'house' ] } },
     { stdout: 'outside of if tiger\noutside of if lion\noutside of if house\ninside of the if\nin if tiger\nin if lion\nin if house\nour categories are:tiger, lion, house, \n',
       func_name: '',
       stack_to_render: [],
       globals: { categoryList: [ 'REF', 1 ] },
       ordered_globals: [ 'categoryList' ],
       line: 21,
       col: 1,
       event: 'step_line',
       heap: { '1': [ 'LIST', 'tiger', 'lion', 'house' ] } } ] }
